### PR TITLE
terraform: describe the nixos.org records in route53

### PR DESCRIPTION
**Pull request by zimbatm**
Right now the DNS is hosted on the udag.org servers and not directly
changeable with code.

### CODE REVIEW

**Comment by zimbatm** [comment]
I would need the full dump from united domains to finish this work. Then we can transition to Route53 for the DNS.

**Comment by zimbatm** [comment]
```
name    ttl IN  type    prio    content
losser.nixos.org.   3600    IN  A       255.255.255.255
*.losser.nixos.org. 3600    IN  A       255.255.255.255
stan.nixos.org. 3600    IN  A       255.255.255.255
*.stan.nixos.org.   3600    IN  A       255.255.255.255
lucifer.nixos.org.  3600    IN  A       255.255.255.255
*.lucifer.nixos.org.    3600    IN  A       255.255.255.255
mturk.nixos.org.    3600    IN  A       130.161.158.181
*.mturk.nixos.org.  3600    IN  A       130.161.158.181
ts.nixos.org.   3600    IN  A       255.255.255.255
*.ts.nixos.org. 3600    IN  A       255.255.255.255
svn.nixos.org.  3600    IN  A       54.217.220.47
planet.nixos.org.   3600    IN  A       54.217.220.47
hydra-test.nixos.org.   3600    IN  A       131.180.119.77
barbrady.nixos.org. 3600    IN  A       144.76.106.45
monitor.nixos.org.  3600    IN  A       144.76.106.45
nixos.org.  3600    IN  A       54.217.220.47
*.nixos.org.    3600    IN  A       54.217.220.47
hydra.nixos.org.    3600    IN  A       46.4.67.10
bastion.nixos.org.  600 IN  A       34.254.208.229
status.nixos.org.   600 IN  A       138.201.32.77
losser.nixos.org.   86400   IN  AAAA        2001:610:685:1:216:3eff:fe00:4
*.losser.nixos.org. 86400   IN  AAAA        2001:610:685:1:216:3eff:fe00:4
stan.nixos.org. 86400   IN  AAAA        2001:610:685:1:222:19ff:fe55:bf2e
*.stan.nixos.org.   86400   IN  AAAA        2001:610:685:1:222:19ff:fe55:bf2e
lucifer.nixos.org.  86400   IN  AAAA        2001:610:685:1:862b:2bff:fe0b:98f0
*.lucifer.nixos.org.    86400   IN  AAAA        2001:610:685:1:862b:2bff:fe0b:98f0
ts.nixos.org.   86400   IN  AAAA        2001:610:685:1:216:3eff:fe01:ee
*.ts.nixos.org. 86400   IN  AAAA        2001:610:685:1:216:3eff:fe01:ee
hydra-test.nixos.org.   86400   IN  AAAA        2001:610:685:1:0:0:0:1
hydra.nixos.org.    3600    IN  AAAA        2a01:4f8:140:248f::
cache.nixos.org.    3600    IN  CNAME       dualstack.v2.shared.global.fastly.net.
releases.nixos.org. 3600    IN  CNAME       d3g5gsiof5omrk.cloudfront.net.
tarballs.nixos.org. 3600    IN  CNAME       d3am6xf9zisc71.cloudfront.net.
*.conf.nixos.org.   3600    IN  CNAME       nixconberlin.github.io.
conf.nixos.org. 3600    IN  CNAME       nixconberlin.github.io.
*.weekly.nixos.org. 3600    IN  CNAME       nixos.github.io.
weekly.nixos.org.   3600    IN  CNAME       nixos.github.io.
discourse.nixos.org.    600 IN  CNAME       nixos1.hosted-by-discourse.com.
_ea68264b3470fb78960575f8dda9b40b.tarballs.nixos.org.   600 IN  CNAME       _d7407b1e66c162385ea6816b6da86f00.acm-validations.aws.
_d8f6310f3e219676be295c56e7084ed2.releases.nixos.org.   600 IN  CNAME       _f66cc632b3b03a0f5493a406c535ad7d.acm-validations.aws.
_d47a77b375708cea087182ee599174c0.cache.nixos.org.  600 IN  CNAME       _f437d0ffb520b017f4d72beb71afedf8.acm-validations.aws.
cache-fastly.nixos.org. 3600    IN  CNAME       v2.shared.global.fastly.net.
nixos.org.  14400   IN  MX  10  mx00.udag.de.
nixos.org.  14400   IN  MX  20  mx01.udag.de.
nixos.org.  86400   IN  NS      ns.udag.org.
nixos.org.  86400   IN  NS      ns.udag.de.
nixos.org.  86400   IN  NS      ns.udag.net.
nixos.org.  86400   IN  SOA     ns.udag.net. hostmaster.united-domains.de 2015072401
```

eelco> "losser",  "stan", "barbrady", "ts", and "mturk" can be removed

**Comment by zimbatm** [comment]
> aws_route53_zone.nixos: AccessDenied: User: arn:aws:iam::080433136561:user/jonas.chevalier is not authorized to perform: route53:CreateHostedZone

File: terraform/route53.tf [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  ttl     = "3600"
+  records = ["dualstack.v2.shared.global.fastly.net"]
+}
+
+resource "aws_route53_record" "nixos-cache-verification" {
+  zone_id = "${aws_route53_zone.nixos.zone_id}"
+  name    = "_d47a77b375708cea087182ee599174c0.cache"
+  type    = "CNAME"
+  ttl     = "600"
+  records = ["_f437d0ffb520b017f4d72beb71afedf8.acm-validations.aws"]
```
Comment (edolstra) [inline terraform/route53.tf]: Do we actually need to keep these after validation has succeeded?

File: terraform/route53.tf [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+resource "aws_route53_record" "nixos-wild" {
+  zone_id = "${aws_route53_zone.nixos.zone_id}"
+  name    = "*"
+  type    = "A"
+  ttl     = "3600"
+  records = ["${local.host_www}"]
+}
+
+## Domain verification
```
Comment (edolstra) [inline terraform/route53.tf]: Stray comment?

File: terraform/route53.tf (hunk 1 above)
Comment (grahamc) [reply terraform/route53.tf]: I think they re-validate every time they re-issue the certificate.

**Comment by grahamc** [comment]
Awesome, assuming it is correct, lgtm :) ;)

File: terraform/route53.tf (hunk 2 above)
Comment (zimbatm) [reply terraform/route53.tf]: yes, removed

File: terraform/route53.tf (hunk 1 above)
Comment (zimbatm) [reply terraform/route53.tf]: I am not sure. It's better to keep them around for now.

**Pull request merged**